pkg/pex/option.go

```diff
@@ -22,12 +22,12 @@ type GossipParams struct {
 type Config struct {
 	fx.Out
 
-	Log          log.Logger
-	Gossip       GossipParams
-	Tick         time.Duration
-	Store        ds.Batching
-	Discovery    discovery.Discovery
-	DiscoveryOpt []discovery.Option
+	Log           log.Logger
+	GossipFactory func(ns string) GossipParams
```

Comment:
I would call these `NewGossip`, `NewTick` (or perhaps `TickFor`?) and `NewStore`, respectively.

The reason is that a "factory" is technically a type, whereas this is effectively a method.